File: robocode2-server/src/main/java/net/robocode2/model/IRound.java
package net.robocode2.model;

import java.util.List;

/**
 * Round interface
 * 
 * @author Flemming N. Larsen
 */
public interface IRound {

	/** Returns the round number */
	int getRoundNumber();

	/** Returns list of turns */
	List<ITurn> getTurns();

	/** Returns flag specifying if round has ended */
	boolean isRoundEnded();

	/**
	 * Returns the last turn of this round
	 */
	default ITurn getLastTurn() {
		List<ITurn> turns = getTurns();
		int numTurns = turns.size();
		if (numTurns > 0) {
			return turns.get(numTurns - 1);
		}
		return null;
	}
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/model/IRound.kt
package dev.robocode.tankroyale.server.model

/** Round interface */
interface IRound {
    /** Round number */
    val roundNumber: Int

    /** List of turns */
    val turns: List<ITurn>

    /** Flag specifying if round has ended yet */
    val roundEnded: Boolean

    /** Last turn */
    val lastTurn: ITurn? get() = if (turns.isNotEmpty()) turns[turns.size - 1] else null;
}
